pycommon/reporter_config/actions/Email.py

```diff
@@ -4,45 +4,80 @@
 from email.mime.text import MIMEText
 import json
 
-__author__ = "Petr Stehlik"
-
 class EmailAction(Action):
     """Action to send an email with IDEA record to a specified address
 
     Mail is send by localhost SMTP server. In the future this
     should be configurable in the config
     """
+    def __init__(self, action):
+        super(type(self), self).__init__(actionId = action["id"], actionType = "email")
 
-    # Cannot use from since it is a keyword
-    __from = "nemea@localhost"
-    msg = None
+        if "to" in a:
+            to = a["to"]
+            self.addrsTo = [email.strip() for email in to.split(",")]
+        else:
+            raise Exception("Email action needs `to` parameter, check your YAML config.")
 
-    def __init__(self, action):
-        self.actionId = action["id"]
-        self.to = action["email"]["to"]
-        self.subject = action["email"]["subject"]
+        if "subject" in a:
+            self.subject = a["subject"]
+        else:
+            raise Exception("Email action needs `subject` parameter, check your YAML config.")
+
+        self.addrFrom   = a["from"]     if "from"     in a else "nemea@localhost"
+        self.smtpServer = a["server"]   if "server"   in a else "localhost"
+        self.smtpPort   = a["port"]     if "port"     in a else 25
+        self.smtpTLS    = a["startTLS"] if "startTLS" in a else False
+        self.smtpSSL    = a["forceSSL"] if "forceSSL" in a else False
+        self.smtpUser   = a["authuser"] if "authuser" in a else None
+        self.smtpPass   = a["authpass"] if "authpass" in a else None
+        key    = a["key"]   if "key"   in a else None
+        chain  = a["chain"] if "chain" in a else None
 
-        self.smtp = smtplib.SMTP('localhost')
+        try:
+            with open(key, "r") as f:
+                self.smtpKey = f.read()
+        except Exception:
+            self.smtpKey = None
```

Comment:
I agree it should be silently skipped if key=None. But if key is set but the file can't be opened (e.g. it doesn't exist), it should be reported.
So, there should be an `if` for the case key=None, but exceptions in file reading should be logged.